business/services.go

```diff
@@ -807,20 +803,30 @@ func (in *SvcService) UpdateService(ctx context.Context, cluster, namespace, ser
 		return nil, fmt.Errorf("cluster: %s not found", cluster)
 	}
 
-	if err := userClient.UpdateService(namespace, service, jsonPatch, patchType); err != nil {
+	svc, err := userClient.UpdateService(namespace, service, jsonPatch, patchType)
+	if err != nil {
 		return nil, err
 	}
 
-	// Stop and restart cache here after a Create/Update/Delete operation to force a refresh
-	// so that the next request that reads from the cache will be sure to have the write operation.
+	in.waitForCacheUpdate(ctx, cluster, svc)
+
+	// After the update we fetch the whole workload
+	return in.GetServiceDetails(ctx, cluster, namespace, service, interval, queryTime, false)
+}
+
+func (in *SvcService) waitForCacheUpdate(ctx context.Context, cluster string, updatedService *core_v1.Service) {
 	kubeCache, err := in.kialiCache.GetKubeCache(cluster)
 	if err != nil {
-		return nil, err
+		log.Errorf("Failed waiting for object to update in cache. You may see stale data but the update was processed correctly. Error: %s", err)
```

Comment:
Yeah, the way this code is, we are going to pull our hair out when we see "Failed waiting for object to update in cache" and have to decide which of the 4 places it was logged :) Unless we actually have structured context in "log" which I don't think this has that yet. So changing the msg to make it unique...

```suggestion
		log.Errorf("Svc Service cannot get cache so cannot wait for object to update in cache. You may see stale data but the update was processed correctly. Error: %s", err)
```